### PR TITLE
[DataGrid] Fix `columnHeadersContainerRef` being `undefined` before mount

### PR DESCRIPTION
**Pull request by samwato**
Closes https://github.com/mui/mui-x/issues/13811

Tests Pass in Demo with PR preview package:
https://codesandbox.io/p/devbox/github/samwato/codesandbox/tree/main/mui-x-issue-13811?file=%2Fsrc%2FDemo.test.tsx%3A12%2C1

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by romgrk — APPROVED** [review]
I think this is a non-breaking API change, though I don't remember why the other non-null changes were done at v7. Maybe because they were grouped with sticky headers.

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md [hunk 1]
```diff
@@ -571,6 +571,7 @@ See the [Direct state access](/x/react-data-grid/state/#direct-selector-access)
 - The field `mainElementRef` is now always non-null.
 - The field `rootElementRef` is now always non-null.
 - The field `virtualScrollerRef` is now always non-null.
+- The field `columnHeadersContainerRef` is now always non-null.
```
Comment (cherniavskii) [inline docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:574]:
```suggestion
```

**Review by cherniavskii — APPROVED** [review]
Thanks @samwato 
Do you need this change ported to v6?
If not, it will only be released in v7.

**Comment by samwato** [comment]
> Thanks @samwato Do you need this change ported to v6? If not, it will only be released in v7.

Was not an issue in v6 so all good.

**Pull request merged**